src/go/configgenerator/cluster_generator.go

```diff
@@ -40,12 +40,27 @@ func MakeClusters(serviceInfo *sc.ServiceInfo) ([]*clusterpb.Cluster, error) {
 		clusters = append(clusters, backendCluster)
 	}
 
-	metadataCluster, err := makeMetadataCluster(serviceInfo)
-	if err != nil {
-		return nil, err
-	}
-	if metadataCluster != nil {
-		clusters = append(clusters, metadataCluster)
+	// When ServiceAccountKey is undefined, envoy will access Instance
+	// Metadata to get access token and identity token. It can be either sidecar
+	// mode and gateway mode.
+	//
+	// When ServiceAccountKey is defined, configmanager will setup a local
+	// server to provide access token generated by the service account credential.
+	// It can only be sidecar mode.
+	if serviceInfo.Options.ServiceAccountKey == "" {
+		metadataCluster, err := makeMetadataCluster(serviceInfo)
+		if err != nil {
+			return nil, err
+		}
+		if metadataCluster != nil {
+			clusters = append(clusters, metadataCluster)
+		}
+
+	} else {
+		latsCluster := makeLatsCluster(serviceInfo)
```

Comment:
Lats is confusing.   Can we call it TokenServer?

Reply:
Admit it is a little bit confusing at the first time but the reason I choose it:
- TokenServer is too general compared to local access token server(the full name of lates). 
- Also, we already use imds for instance metadata server and we clarify in the comments, which is the similar case.

Reply:
OK, then.  I just need to get used to this name.